frontend/src/components/FacilityBooking/BookingBlock.tsx

```diff
@@ -13,38 +13,38 @@ type Props = {
 }
 
 const BookingBlock = (props: Props) => {
-  const { selectedStartTime } = useSelector((state: RootState) => state.facilityBooking)
   const ref = useRef<HTMLDivElement>(null)
 
+  const { selectedStartTime, selectedEndTime } = useSelector((state: RootState) => state.facilityBooking)
+
   useEffect(() => {
     if (props.scrollTo) {
       scrollToView(ref)
     }
   }, [ref.current])
 
   function blockText() {
-    let text = ''
     switch (props.entry.type) {
       case TimeBlockType.OCCUPIED:
-        text = 'Occupied'
-        break
+        return 'Occupied'
       case TimeBlockType.AVAILABLE:
-        text = 'Available'
-        break
+        return 'Available'
       case TimeBlockType.SELECTED:
-        text = 'Selected as starting time'
-        break
+        return 'Selected as starting time'
       default:
-        break
+        return ''
     }
-    return text
   }
 
   function hasOverlay() {
     const currentTimestamp = Math.round(Date.now() / 1000)
     const currentStartHour = currentTimestamp - (currentTimestamp % 3600)
```

Comment:
Can consider using `getHour()`

Reply:
i think it's fine cos dealing with timestamp is more straightforward